bin/compile

```diff
@@ -34,49 +37,69 @@ export_env_dir "$3"
 # Create the cache directory if not exist.
 mkdir -p $CACHE_DIR
 
-cd $BUILD_DIR
+# Get the source dir
+if [ -n "$FLUTTER_SOURCE_DIR" ]; then
+  SOURCE_DIR=$BUILD_DIR/$FLUTTER_SOURCE_DIR
+else
+  SOURCE_DIR=$BUILD_DIR
+fi
+cd "$SOURCE_DIR"
 
-if [ -d "$CACHE_DIR/flutter" ]; then
+# Get the deploy build
+if [ -n "$FLUTTER_DEPLOY_DIR" ]; then
+  DEPLOY_DIR=$BUILD_DIR/$FLUTTER_DEPLOY_DIR
+else
+  DEPLOY_DIR=$BUILD_DIR/public_html
+fi
+mkdir -p $DEPLOY_DIR
+
+# Restore Flutter SDK from cache
+if [ -d "$CACHE_DIR/.flutter" ]; then
   print "Restoring Flutter SDK from CACHE"
-  cp -R $CACHE_DIR/flutter $BUILD_DIR
+  cp -R $CACHE_DIR/.flutter $SOURCE_DIR
+  cp -R $CACHE_DIR/.pub-cache $SOURCE_DIR
 else
   print "Installing SDK from Github repository."
-  git clone https://github.com/flutter/flutter.git --quiet
+  git clone https://github.com/flutter/flutter.git .flutter --quiet
```

Comment:
Usually this would an excellent idea, but the `flutter version` command uses git tags for switching between versions, so in this case the history is necessary.